app/src/main/java/com/example/taskyy/data/repositories/AuthRepositoryImpl.kt

```diff
@@ -56,17 +45,31 @@ class AuthRepositoryImpl @Inject constructor(
     override suspend fun login(login: Login): Result<User> {
         return try {
             val user = retrofit.loginUser(login)
+
+            if (!userDao.doesUserExist(user.userId)) {
+                val userEntity = user.toUserEntity(login.email, user.userId)
+                userDao.insertUser(userEntity)
+            }
```

Comment:
To avoid needing to check if the user exists, you can also annotate your insert function with `@Upsert`. That will insert a new entry or update the existing one if there is already one with the ID you're trying to insert